File: Art Resources/resource/DungeonTileset/script_example.lua
-- prints to stdout file ready to be `require`d in your game

local file = arg[1]

if not file then
  print("usage: lua " .. arg[0] .. " ./tiles_list")
  os.exit()
end

local data = {}
local fmt = '    %s = { x=%d, y=%d, w=%d, h=%d, animLen=%d, isAnim=%s },'
local tn = tonumber
local fp, err = io.open(arg[1], 'r')

if not fp then
  print(err)
  os.exit()
end

local pattern = '([a-z_]+)' .. (' ([0-9]+)'):rep(4) .. ' *([0-9+]?)'
for line in fp:lines() do
  --local name, x, y, w, h, anim_len = line:match('([a-z_]+)')
  local name, x, y, w, h, animLen = line:match(pattern)

  if name then
    data[#data+1] = fmt:format(
      name, tn(x), tn(y), tn(w), tn(h), tn(animLen == '' and 1 or animLen), animLen ~= '')
  end
end
fp:close()

table.sort(data)

io.write('return {\n', table.concat(data, '\n'), '\n}\n')

File: Art Resources/resource/DungeonTileset/script_example.js
// If all goes well prints JSON to stdout

const fs = require('fs');
const readline = require('readline');
const path = process.argv[process.argv.length-1];

let data = {};
var rl = readline.createInterface({
  input : fs.createReadStream(path),
  output: process.stdout,
  terminal: false,
});
rl
.on('line', (line) => {
  let m = line.match(/([a-z_]+) +([0-9]+) +([0-9]+) +([0-9]+) +([0-9]+) ?([0-9]?)/);
  if(m){
    data[m[1]] = {
      x: parseInt(m[2]),
      y: parseInt(m[3]),
      w: parseInt(m[4]),
      h: parseInt(m[5]),
      animLen: parseInt(m[6] || 1),
      isAnim: !!m[6],
    };
  }
})
.on('close', () => {
  console.log(JSON.stringify(data, null, 4));
});

